clang/lib/Tooling/DependencyScanning/DependencyScanningWorker.cpp

```diff
@@ -393,8 +393,6 @@ class DependencyScanningAction {
                      DiagnosticConsumer *DiagConsumer) {
     // Make a deep copy of the original Clang invocation.
     CompilerInvocation OriginalInvocation(*Invocation);
-    if (any(Service.getOptimizeArgs() & ScanningOptimizations::Macros))
-      canonicalizeDefines(OriginalInvocation.getPreprocessorOpts());
```

Comment:
I'd probably just move these up and pass in `Invocation` instead of `OriginalInvocation`.